deployment/desktop/desktop-application-metadata-overview.md

```diff
@@ -0,0 +1,155 @@
+
+If the application is being built on your own machines or cloud instance then the distributor will need to generate the AppStream metadata manually. This would for example be the case when internal-only or closed source software is being either used or produced. This document assumes you are currently building RPM packages and exporting yum-style repository metadata for Fedora or RHEL although the concepts are the same for rpm-on-OpenSuse or deb-on-Ubuntu.
+
+NOTE: If you are building packages, make sure that there are not two applications installed with one single package. If this is currently the case split up the package so that there are multiple subpackages or mark one of the *.desktop* files as *NoDisplay=true*. Make sure the application-subpackages depend on any *-common* subpackage and deal with upgrades (perhaps using a metapackage) if you’ve shipped the application before.
+
+### Summary of Package building
+
+The steps outlined above explain the extra metadata that you need for your application to show up in GNOME Software. This tutorial does not cover how to set up your build system to build these, but both for  Meson and autotools you should be able to find a plethora of examples online.
+
+There are also major resources available to explain how to create a [Fedora RPM](https://fedoraproject.org/wiki/How_to_create_an_RPM_package) or [how to build a Flatpak](http://docs.flatpak.org/en/latest/). You probably also want to tie both the Desktop file and the AppData file into your i18n system so the metadata in them can be translated.
+
+It is worth noting here that, while this document explains how you can do everything yourself, we generally recommend relying on existing community infrastructure for hosting source code and packages if you can (for instance, if your application is open source). They will save you work and effort over time. For example, putting your source code on GNOME git will make it easy for the GNOME translation teams to work on your application and thus increase the chance that it gets translated in many languages. And by building your package in Fedora you can get peer review of your package and free hosting of the resulting package. We are also working on a service that will automatically generate a Flatpak bundle of your application in Fedora, meaning you get a Flatpak version of your application with no extra effort.
+
+### Citations
+
+1. [Filesystem Hierarchy Standard: 17th October 2016](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard) 
+2. [Desktop Specification: 17th October 2016](https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#introduction) 
+3. [AppStream Specification 0.10: 17th October 2016](https://www.freedesktop.org/software/appstream/docs/) 
+4. [SPDX license list](https://spdx.org/licenses/)
```

Comment:
No citations needed here. This is not a formal document.

There should be handy links that user might actually need to get more info, sources etc.